### DISCUSSION TITLE
Using a remark plugin with react-markdown

### DISCUSSION THREAD

**Discussion by benel** [remarkjs/remark#1110] [Q&A] [answered]
Dear "remarkers",

First, thank you for the great ecosystem you provide.

I have a question about using a remark plugin ([remark-definition-list](https://github.com/wataru-chocola/remark-definition-list#remark-definition-list)) with `remarkjs/react-markdown`. I am not sure about the way to combine the instructions given in both documentations.

The plugin documentation states that it should be used this way:

```js
const html = unified()
  .use(remarkParse)
  .use(remarkDefinitionList)
  .use(remarkRehype, {
    handlers: {
      // any other handlers
      ...defListHastHandlers,
    }
  })
  .use(rehypeStringify)
  .process(md);
```

Reading the documentation of  `remarkjs/react-markdown`, I tried this:
```js
return (
  <ReactMarkdown children={md}
    remarkPlugins={[remarkDefinitionList]} 
    rehypePlugins={[defListHastHandlers]}
  />
);
```

Is this really the equivalent way to call these plugins in `remarkjs/react-markdown`?

When I test it on a markdown definition list, I get `div`s instead. :'(

**Comment by remcohaszing** [comment, marked as answer]
> Dear "remarkers",

That’s a _remarkable_ pun!

`defListHastHandlers` is not a rehype plugin. They are options for the `remark-rehype` plugin that’s already used internally by `react-markdown`. It’s should be passed as a `remarkRehypeOptions`.

```jsx
return (
  <ReactMarkdown
    remarkPlugins={[remarkDefinitionList]} 
    remarkRehypeOptions={{
      handlers: {
        // any other handlers
        ...defListHastHandlers
      }
    }}
  >
    {md}
  </ReactMarkdown>
);
```

It’s also recommended to pass React children as actual children, not using a prop.

Is possible, you may want to move `remarkRehypeOptions` outside of your component or memoize them to prevent unnecessary rerenders.

**Reply by benel** [reply]
Thank you very much for your answer! I will try that.

**Reply by benel** [reply]
It works! Thank you very much!
